code/game/objects/effects/anomalies/anomaly_bleed.dm

```diff
@@ -0,0 +1,60 @@
+/obj/effect/anomaly/blood
+	name = "blood anomaly"
+	icon_state = "blood_anomaly"
+	density = TRUE
+	aSignal = /obj/item/assembly/signaler/anomaly/blood
+	var/sucking = FALSE
+
+/obj/effect/anomaly/blood/anomalyEffect(delta_time)
+	if (sucking)
+		return
+	..()
+	if (DT_PROB(20, delta_time))
+		AddComponent(/datum/component/pellet_cloud, projectile_type=/obj/projectile/bullet/shrapnel/bleed, magnitude=3)
+		playsound(src, 'sound/weapons/shrapnel.ogg', 70, TRUE)
+	if (DT_PROB(10, delta_time))
+		for (var/mob/living/carbon/bleed_target in view(6, src))
+			if (!bleed_target.resting)
+				continue
+			if (do_teleport(src, get_turf(bleed_target), channel = TELEPORT_CHANNEL_BLUESPACE))
+				bleed_target.Stun(3 SECONDS)
+				INVOKE_ASYNC(src, PROC_REF(suck_blood))
+			break
+
+/obj/effect/anomaly/blood/detonate()
+	STOP_PROCESSING(SSobj, src)
+	// Needs to sleep since this gets instantly deleted as soon as the proc ends
+	for (var/mob/living/carbon/human/player in GLOB.player_list)
+		if (!is_station_level(player.z))
+			continue
+		var/turf/player_loc = get_turf(player)
+		var/list/nearby_turfs = RANGE_TURFS(4, player_loc)
+		shuffle_inplace(nearby_turfs)
+		var/turf/target = locate(/turf/open) in nearby_turfs
+		if (!target)
+			continue
+		// Blocked by the bluespace anchor
+		if (!do_teleport(src, target, channel = TELEPORT_CHANNEL_BLUESPACE))
+			return
+		sleep(40)
+		if (QDELETED(src))
+			return
+		AddComponent(/datum/component/pellet_cloud, projectile_type=/obj/projectile/bullet/shrapnel/bleed, magnitude=3)
+		playsound(src, 'sound/weapons/shrapnel.ogg', 70, TRUE)
+		sleep(20)
+		if (QDELETED(src))
+			return
+
+/obj/effect/anomaly/blood/proc/suck_blood()
+	sucking = FALSE
+	for (var/mob/living/carbon/bleed_target in loc)
+		bleed_target.add_bleeding(BLEED_SURFACE)
+		bleed_target.emote("scream")
+		sucking = TRUE
+	if (sucking)
+		new /obj/effect/temp_visual/cult/sparks(loc)
+		addtimer(CALLBACK(src, PROC_REF(suck_blood)), 0.5 SECONDS)
```

Comment:
I don't think it's reachable logic.

Reply:
It is via line 53

Reply:
oh, you're right, but I still feel this off.
`suck_blood()` is async'ed, and this stacks up for every 0.5s., and it won't gone itself until there's no victim. So, this can exist infinite proc calls in a worst case... if it's intented.

Reply:
This can't stack because the global variable sucking is set to true, which prevents the proc chain being called more than once at a time.

Reply:
It will be true everytime it's called nearby mobs by line 53. no?

Reply:
It will be set to false when its finished